### PR TITLE
Fix issue of saving partial index when index is gpu based

### PR DESCRIPTION
**Pull request by oneraghavan**
## Why ?

To Fix issue of saving partial index when index is gpu based

Why do we need to implement this feature ? What is the use case ?

## How ?

Document the technical decisions you made.
If some parts are WIP, please explicit them here.

## Test plan

How did you test your changes ?
Include full command line to help other people reproduce if needed.

### CODE REVIEW

**Comment by oneraghavan** [comment]
Fixing this issue #5

File: stopes/modules/bitext/indexing/populate_faiss_index.py [hunk 1]
```diff
@@ -289,8 +289,10 @@ def add_embedding_to_index(
                 # but is quite slow, and when we aren't afraid of preemption we should skip it.
                 # Write partial index to the partial index file (and while this happens, keep is_partial_file_valid as False)
                 checkpoint_summary.is_partial_file_valid = False
+                if gpu:
+                    partial_cpu_idx = faiss.index_gpu_to_cpu(partial_idx)
                 faiss.write_index(
-                    partial_idx,
```
Comment (Mortimerp9) [inline stopes/modules/bitext/indexing/populate_faiss_index.py:293]: That's not going to work if it's not on GPU as partial_cpu_index is undefined unless it goes through the if.

Comment (oneraghavan) [reply stopes/modules/bitext/indexing/populate_faiss_index.py:293]: Done

**Review by Mortimerp9 — APPROVED** [review]

**Comment by Mortimerp9** [comment]
@oneraghavan thank you so much for this fix. This is the first public PR on the repo, thanks! 

Could you check the lint failure? It looks like you need to apply `black` on the file (the command is in the logs of the check).

**Comment by Mortimerp9** [comment]
black still seems unhappy. Do you mind if I push a fix to your PR?

**Pull request merged**